doc/ondemand.md

```diff
@@ -37,31 +37,46 @@ into your project. Then include it in your project with:
 ```c++
 #include "simdjson.h"
 using namespace simdjson; // optional
-using namespace simdjson::builtin; // optional, for ondemand
+using namespace simdjson::builtin; // optional, for On Demand
 ```
 
-You can compile with:
+You can generally compile with:
 
 ```
 c++ -march=native myproject.cpp simdjson.cpp
 ```
 
+
 Note:
+- The `-march=native` is only needed under x64 systems (Intel/AMD).
+- CMake users who wish to pass the `-march=native` flag may modify the `CMAKE_CXX_FLAGS` flag:  e.g., `cmake  -DCMAKE_CXX_FLAGS="-march=haswell"  ...`.
 - Users on macOS and other platforms where compilers do not provide C++11 compliant by default
 should request it with the appropriate flag (e.g., `c++ -march=native -std=c++17 myproject.cpp simdjson.cpp`).
 
-### The native architecture flag
+### The native architecture flag (x64 systems)
 
-Passing `-march=native` to the compiler makes On Demand much faster by allowing it to use
-optimizations specific to your machine. You cannot do this, however, if you are compiling code
-that might be run on less advanced machines.
 
-On Demand uses advanced architecture-specific code for many common processors to make JSON
+The On Demand API uses advanced architecture-specific code for many common processors to make JSON
 preprocessing and string parsing faster. By default, however, most c++ compilers will compile to the
 least common denominator (since the program could theoretically be run anywhere). Since On Demand is
 inlined into your own code, it cannot use these advanced versions unless the compiler is told to
-target them. -march=native says "target the current computer," which is a reasonable default for
+target them. The `-march=native` flags says "target the current computer," which is a reasonable default for
 many applications which both compile and run on the same processor.
+The `-march=native` flag is unsupported and unnecessary on some platforms such as ARM (aarch64) but useful 
+for best performance on x64 (e.g., Intel) systems. 
+
+Passing `-march=native` to the compiler may make On Demand much faster by allowing it to use
```

Comment:
I actually think we can remove -march=native from the recommendation here, now that we've switched to using a runtime-selected stage 1. Once we do that, we can move this recommendation and caution to performance.md, which will shorten this quick start a lot, making it much more, you know, quick :) It's for a future checkin though.

We also can revisit both quickstarts and add -O3; while it's absolutely true that we compile fine without it, we should start people out on the right foot. This is probably a change from my past position :)

Reply:
Removed `-march=native` and added `-O3`.